Sources/NIOHTTP1/HTTPPipelineSetup.swift

```diff
@@ -53,20 +54,112 @@ extension ChannelPipeline {
     public func addHTTPClientHandlers(position: Position = .last,
                                       leftOverBytesStrategy: RemoveAfterUpgradeStrategy = .dropBytes,
                                       withClientUpgrade upgrade: NIOHTTPClientUpgradeConfiguration?) -> EventLoopFuture<Void> {
-        
+        let future: EventLoopFuture<Void>
+
+        if self.eventLoop.inEventLoop {
+            future = self.eventLoop.makeCompletedVoidFuture {
+                return try self.syncOperations.addHTTPClientHandlers(position: position,
+                                                                     leftOverBytesStrategy: leftOverBytesStrategy,
+                                                                     withClientUpgrade: upgrade)
+            }
+        } else {
+            let promise = self.eventLoop.makePromise(of: Void.self)
+            future = promise.futureResult
+
+            self.eventLoop.execute {
```

Comment:
This seems like a somewhat complex spelling of `eventLoop.submit`, isn't it?

Reply:
🤦‍♂️